common/network_manager.go

```diff
@@ -394,14 +401,21 @@ func (m *NetworkManager) onLoadingFinished(event *network.EventLoadingFinished)
 		}
 	}
 	req.responseEndTiming = float64(event.Timestamp.Time().Unix()-req.timestamp.Unix()) * 1000
+	m.deleteRequestByID(event.RequestID)
+	m.frameManager.requestFinished(req)
+
 	// Skip data and blob URLs when emitting metrics, since they're internal to the browser.
 	if !isInternalURL(req.url) {
-		req.responseMu.RLock()
-		m.emitResponseMetrics(req.response, req)
-		req.responseMu.RUnlock()
+		// We need to process requestPaused messages from CDP in order to get the response
+		// for the request. However, we can't process them until the request is unblocked.
+		// Since we're blocking the NetworkManager goroutine here, we need to spawn a new
+		// goroutine to allow the requestPaused messages to be processed by the NetworkManager.
+		go func() {
```

Comment:
An alternative is to do this only if the interception is enabled. But I believe emitting response metrics can be async. Maybe we should add a `Context` here to timeout.